src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,459 @@
+                if 0 != noise_reduction:
+                    low_mask = (sum < 0)
+                    sum[low_mask] = 0  # underflow?
+
+                high_mask = (sum > maxlevel)
+                sum[high_mask] = maxlevel  # saturated?  for max image==1.0
+
+                imgout[:, j] = sum
+            else:
+                imgout[:, j] = imgin[:, xindexL[j]] * xtable.bgfactorL
+        else:
+            imgout[:, j] = xtable.mean_modevalue * xtable.bgfactorL
+
+    # Right side
+    for j in range(0, (int(aR))):
+        sum[:] = 0.0
+        if xindexR[j] >= 0:
+            if xindexRB[j - 1] >= 0.0:
+                s = int(math.floor(xindexRB[j - 1]))
+                e = int(math.floor(xindexRB[j]))
+
+                sum[:] = (
+                    (s + 1 - xindexRB[j - 1]) * imgin[:, 511 - s]
+                    + (xindexRB[j] - e) * imgin[:, 511 - e]
+                )
+
+                if (e-s) > 1:  # have a middle pixel
+                    sum[:] = sum[:] + imgin[:, 511 - (s + 1)]
+
+                # Perform the desired noise reduction method
+                if 0 == noise_reduction:
+                    pass
+
+                elif 1 == noise_reduction:
+                    nf = 2 * stdv_modevalue * (xindexRB[j] - xindexRB[j-1] - 1)
+                    sum = sum - nf
+
+                elif 2 == noise_reduction:
+                    nf = 1 * stdv_modevalue * (xindexRB[j] - xindexRB[j-1] - 1)
+                    sum = (sum / (xindexRB[j] - xindexRB[j-1])) - nf
+
+                elif 3 == noise_reduction:
+                    sum = sum / (xindexRB[j] - xindexRB[j-1])
+
+                else:
+                    low_mask = None
+                    high_mask = (sum > maxlevel)
+
+                # TODO: Check on this. Make sure it wasn't an
+                # error in the orginal code
+                if 0 != noise_reduction:
+                    low_mask = (sum < 0)
+                    sum[low_mask] = 0  # underflow?
+
+                high_mask = (sum > maxlevel)
+                sum[high_mask] = maxlevel  # saturated?  for max image==1.0
+
+                imgout[:, j] = sum
+            else:
+                imgout[:, 511 - j] = (
+                    imgin[:, 511 - xindexR[j]]
+                    * xtable.bgfactorR
+                )
+        else:
+            imgout[:, 511 - j] = xtable.mean_modevalue * xtable.bgfactorR
+
+    if FOVwidth == 512:
+        img = imgout.astype(np.uint16)
+    else:
+        img = imgout[:, xtable.L:512 - xtable.R].astype(np.uint16)
+
+    return img
+
+
+def create_xtable(meanimg, stdvimg, aL, aR, bL, bR, noise_reduction):
+    """
+
+    """
+    # assume flat area aL - (512-aR)
+
+    xindexL = np.zeros(256, np.int)
+    xindexLB = np.zeros(256, np.float)  # between pixels
+    xindexR = np.zeros(256, np.int)
+    xindexRB = np.zeros(256, np.float)  # between pixels
+
+    # Left side
+    for j in range(0, int(aL)):
+        xindexL[j] = (
+            j - int(
```

Comment:
These look really repetitive... can we factor out into a funtion?

Reply:
Yep, done now!